### PR TITLE
Add gradient_accumulation_steps to pretrain/train API

### LINKED ISSUE #35: [FEAT] Gradient accumulation
### 💡 Is your feature request related to a problem?

Feature request.When training on GPUs with limited VRAM, users are forced to use small batch sizes. Gradient accumulation would allow users to achieve equivalent large batch sizes while working within hardware constraints.

### 🧰 Describe the solution you'd like

Add a `gradient_accumulation_steps` parameter to the `lightly_train.train()` function:  

This should:  
1. Accumulate gradients over N forward/backward passes  
2. Only perform optimizer step/weight update every N steps  
3. Automatically scale loss values if needed  

### 📝 Additional context

**Technical Implementation**:  
- Add gradient accumulation logic in optimizer step:  
```python
if (step + 1) % gradient_accumulation_steps == 0:
    optimizer.step()
    optimizer.zero_grad()
```  

**Example Usage**:  
```python
# Effective batch size = 32 (8 per step * 4 accumulation steps)
lightly_train.train(
    # ... other params ...
    batch_size=8,
    gradient_accumulation_steps=4
)
```  

**Benefits**:  
- No VRAM overhead - same memory usage as baseline batch size  
- Backwards compatible (default=1 maintains original behavior)  
- Supported by most PyTorch optimizers natively  
- Common pattern in other ML libraries (HuggingFace, PyTorch Lightning)

### PR DESCRIPTION
**Pull request by tetelias**
### Summary

Adds `gradient_accumulation_steps` to the pretrain/train API as a convenience alias for PyTorch Lightning's `accumulate_grad_batches`.

This makes the pretraining API more consistent with the task-specific training APIs while preserving the existing `trainer_args` escape hatch.

### Changes

- add `gradient_accumulation_steps` parameter
- map to `Trainer(accumulate_grad_batches=...)`
- add validation/conflict handling
- add tests

### Example

```python
lightly_train.pretrain(
    ...,
    batch_size=8,
    gradient_accumulation_steps=4,
)
```
Equivalent to:
```python
lightly_train.pretrain(
    ...,
    trainer_args={
        "accumulate_grad_batches": 4,
    },
)
```

### Reasoning
This resolves https://github.com/lightly-ai/lightly-train/issues/35

## How has it been tested?

In tests/_commands/test_train_helpers.py test_get_trainer was updated, test_get_trainer_gradient_accumulation and test_get_trainer_gradient_accumulation_conflict were added.
All of: 
tests 
ruff check . 
ruff format . 
pre-commit run --all-files 
pass without errors.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

**Comment by liopeer** [comment]
Thanks for the contribution @tetelias! Could you sign the CLA?

**Comment by tetelias** [comment]
@liopeer  I signed CLA and corrected failure on one of tests. Do you need to restart workflows approval?

**Review by liopeer — COMMENTED** [review]
Thanks, that's really high quality work. If all the checks succeed, this is ready to merge!

**Review by liopeer — APPROVED** [review]
LGTM!

**Comment by liopeer** [comment]
@tetelias It looks like there are still issues with the CLA. Can you quickly check again if it is really signed?

**Comment by DLemming** [comment]
One thought: should `accumulate_grad_batches` also be included when computing `global_batch_size`?

Right now it's:

`global_batch_size = args.batch_size * args.devices`

but the effective batch size is really:

`effective_batch_size = args.batch_size * args.devices * args.acc_grad_batches`

Since `global_batch_size` is used for automatic LR scaling, exposing `accumulate_grad_batches` without accounting for it means users get a larger effective batch size but the LR is still scaled for the smaller one. As lightly-train is a high-level, worry-free wrapper around lightly, as a user I would expect to be accounted for correct learning rate scaling when simulating larger effective batch sizes.

One caveat though, `global_batch_size` is reused for non-LR purposes (e.g. `steps_per_epoch = dataset_size // self.global_batch_size` in `dinov2.py:675` or throughput logging). So multiplying globally will have side effects and it's probably safer to pass through `gradient_accumulation_steps` to the lr_scale line specifically.

**Comment by liopeer** [comment]
Thanks @DLemming, completely right & I had missed this. It's now addressed!
